### PR TITLE
API deprecate -p shortcut in favor of -o for Objective

### LINKED ISSUE #225: CLI: could -p be -o ?
-o for Objective, as -s for Solver and -d for Dataset

### PR DESCRIPTION
**Pull request by mathurinm**
fixes #225 
I have an issue, the warnings are captured even with BENCHOPT_DEBUG=1, I think they should not because the user won't be notified of the deprecation.

It remains to adapt the test if you agree with this @tomMoral @agramfort

### CODE REVIEW

File: benchopt/cli/main.py [hunk 1]
```diff
@@ -93,7 +97,14 @@
 def run(benchmark, solver_names, forced_solvers, dataset_names,
         objective_filters, max_runs, n_repetitions, timeout,
         plot=True, html=True, pdb=False, do_profile=False,
-        env_name='False'):
+        env_name='False', old_objective_filters=None):
+    if len(old_objective_filters):
+        print("should warn but nothing is printed")
+        warnings.warn(
```
Comment (mathurinm) [inline benchopt/cli/main.py:103]: @agramfort do you know why the warning is captured ? The print is executed if you use `-p` but the warning is not displayed

**Comment by agramfort** [comment]
maybe a call to catch_warnings somewhere?

**Comment by mathurinm** [comment]
@tomMoral do you know where this is coming from ? If we call `warnings.resetwarnings()` of `warnings.setfilter("always")` before `warnings.warn`, then the warnings are shown (in addition it shows useful DeprecationWarning for `stping_strategy`). 

So something is catching the warnings somewhere ?

**Review by agramfort — COMMENTED** [review]
this requires a what's new API change entry.

no doc to be updated?

**Comment by mathurinm** [comment]
I added the whatsnew entry, the doc is OK because it's generated from the python click file and there are no examples where we filter objectives

**Review by agramfort — APPROVED** [review]

**Pull request merged**